tests/cpp/common/test_hist_util.cu

```diff
@@ -210,6 +224,19 @@ TEST(HistUtil, DeviceSketchMultipleColumnsExternal) {
   }
 }
 
+template <typename Adapter>
+void ValidateBatchedCuts(Adapter adapter, int num_bins, int num_columns, int num_rows,
+                         DMatrix* dmat) {
+  common::HistogramCuts batched_cuts;
+  SketchContainer sketch_container(num_bins, num_columns, num_rows);
+  AdapterDeviceSketch(adapter.Value(), num_bins, std::numeric_limits<float>::quiet_NaN(),
```

Comment:
This interface should return the entire sketch not some partially computed version.

Reply:
See https://github.com/dmlc/xgboost/pull/5760#discussion_r436484897 .